### PR TITLE
test/kafka-exactly-once: convert to mzworkflows.py

### PR DESCRIPTION
**Pull request by benesch**
Touches #9809.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: test/kafka-exactly-once/mzworkflows.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    parser = WorkflowArgumentParser(w)
+    parser.add_argument(
+        "--seed",
+        help="an alternate seed to use to avoid clashing with existing topics",
+        type=int,
+        default=1,
+    )
+    args = parser.parse_args(args)
+
+    w.start_services(services=["kafka", "schema-registry", "materialized"])
```
Comment (philip-stoev) [inline test/kafka-exactly-once/mzworkflows.py:41]: I do not think this is safe as in rare cases some of the Java services will fail to start if the previous ones in the sequence have not started already.

We have `w.start_and_wait_for_tcp` that takes a list of actual service objects and starts them **in sequence**, pinging each other as it goes. This is safer than starting the Java parts with start_services as start_services does not guarantee that the previous service has started already before the next one is started.

File: test/kafka-exactly-once/mzworkflows.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    w.wait_for_tcp(host="kafka", port=9092)
+    w.wait_for_tcp(host="schema-registry", port=8081)
+    w.wait_for_mz()
+    w.run_service(
+        service="testdrive-svc",
+        command=f"--seed {args.seed} --kafka-option=group.id=group1 before-restart.td",
```
Comment (philip-stoev) [inline test/kafka-exactly-once/mzworkflows.py:47]: The seed can be supplied directly as an argument to the `Testdrive()` constructor.

**Review by philip-stoev — APPROVED** [review]
LGTM, except for the startup sequence around the Java services.

File: test/kafka-exactly-once/mzworkflows.py (hunk 1 above)
Comment (benesch) [reply test/kafka-exactly-once/mzworkflows.py:41]: Ooh, thanks!

**Comment by benesch** [comment]
Fixed the thing you pointed out and kicked off a nightly run: https://buildkite.com/materialize/nightlies/builds/556. Thanks!

**Comment by benesch** [comment]
Oh, no, what am I thinking; this runs against PRs, not nightly.

File: test/kafka-exactly-once/mzworkflows.py (hunk 2 above)
Comment (benesch) [reply test/kafka-exactly-once/mzworkflows.py:51]: Unfortunately doing it that way means you can't pass it in as a command-line argument!

**Pull request merged**